hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -591,4 +596,24 @@ private static void findAnyMissingFieldsRec(Schema incomingSchema,
   public static String createSchemaErrorString(String errorMessage, Schema writerSchema, Schema tableSchema) {
     return String.format("%s\nwriterSchema: %s\ntableSchema: %s", errorMessage, writerSchema, tableSchema);
   }
+
+  /**
+   * Create a new schema by change the nullabilities of the given columns.
+   *
+   * @param schema original schema
+   * @param nullableUpdateCols columns that would be updated with given nullability
+   * @param nullable nullability of column type
+   * @return a new schema with the nullabilities of the given columns updated
+   */
+  public static Schema createSchemaWithNullabilityUpdate(
```

Comment:
Rename to `forceNullableColumns`, we should also avoid to recreate the schema if the field is already nullable.